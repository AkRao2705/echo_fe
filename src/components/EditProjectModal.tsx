import { useState } from 'react';

interface EditProjectModalProps {
  project: { id: string; name: string; description: string };
  onClose: () => void;
  onUpdate: (updatedProject: { id: string; name: string; description: string }) => void;
}

export function EditProjectModal({ project, onClose, onUpdate }: EditProjectModalProps) {
  const [name, setName] = useState(project.name);
  const [description, setDescription] = useState(project.description);

  const handleSubmit = () => {
    onUpdate({ id: project.id, name, description });
  };

  return (
    <div className="fixed inset-0 bg-gray-600 bg-opacity-50 flex justify-center items-center">
      <div className="bg-white p-6 rounded-lg shadow-lg">
        <h2 className="text-lg font-semibold">Edit Project</h2>
        <input
          type="text"
          className="border p-2 w-full my-2"
          value={name}
          onChange={(e) => setName(e.target.value)}
        />
        <textarea
          className="border p-2 w-full my-2"
          value={description}
          onChange={(e) => setDescription(e.target.value)}
        />
        <div className="flex justify-end gap-3">
          <button className="px-4 py-2 bg-gray-400 rounded" onClick={onClose}>Cancel</button>
          <button className="px-4 py-2 bg-blue-600 text-white rounded" onClick={handleSubmit}>Save</button>
        </div>
      </div>
    </div>
  );
}
