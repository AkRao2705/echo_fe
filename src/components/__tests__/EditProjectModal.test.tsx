import { render, screen, fireEvent } from '@testing-library/react';
import { EditProjectModal } from '../EditProjectModal';

describe('EditProjectModal', () => {
  const mockProject = {
    id: '123',
    name: 'Test Project',
    description: 'Test Description'
  };

  const mockOnClose = jest.fn();
  const mockOnUpdate = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('matches snapshot', () => {
    const { container } = render(
      <EditProjectModal
        project={mockProject}
        onClose={mockOnClose}
        onUpdate={mockOnUpdate}
      />
    );
    
    expect(container).toMatchSnapshot();
  });

  it('renders with initial project data', () => {
    // ... rest of the existing tests ...
  });
}); 