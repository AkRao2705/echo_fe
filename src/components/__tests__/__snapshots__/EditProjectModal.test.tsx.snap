// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditProjectModal matches snapshot 1`] = `
<div>
  <div
    class="fixed inset-0 bg-gray-600 bg-opacity-50 flex justify-center items-center"
  >
    <div
      class="bg-white p-6 rounded-lg shadow-lg"
    >
      <h2
        class="text-lg font-semibold"
      >
        Edit Project
      </h2>
      <input
        class="border p-2 w-full my-2 text-gray-900"
        type="text"
        value="Test Project"
      />
      <textarea
        class="border p-2 w-full my-2 text-gray-900"
      >
        Test Description
      </textarea>
      <div
        class="flex justify-end gap-3"
      >
        <button
          class="px-4 py-2 bg-gray-400 rounded"
        >
          Cancel
        </button>
        <button
          class="px-4 py-2 bg-blue-600 text-white rounded"
        >
          Save
        </button>
      </div>
    </div>
  </div>
</div>
`;
